lib/logstash/inputs/s3.rb

```diff
@@ -91,189 +129,165 @@ def register
+      return true
+    else
+      return false
+    end
+  end
 
   private
   def process_log(queue, key)
-
     object = @s3bucket.objects[key]
-    tmp = Dir.mktmpdir("logstash-")
-    begin
-      filename = File.join(tmp, File.basename(key))
-      File.open(filename, 'wb') do |s3file|
-        object.read do |chunk|
-          s3file.write(chunk)
-        end
-      end
-      process_local_log(queue, filename)
-      unless @backup_to_bucket.nil?
-        backup_object = @backup_bucket.objects[key]
-        backup_object.write(Pathname.new(filename))
-      end
-      unless @backup_to_dir.nil?
-        FileUtils.cp(filename, @backup_to_dir)
-      end
-      if @delete
-        object.delete()
-      end
-    end
-    FileUtils.remove_entry_secure(tmp, force=true)
 
-  end # def process_log
+    tmp = Stud::Temporary.directory("logstash-")
 
-  private
-  def process_local_log(queue, filename)
+    filename = File.join(tmp, File.basename(key))
 
-    metadata = {
-      :version => nil,
-      :format => nil,
-    }
-    File.open(filename) do |file|
-      if filename.end_with?('.gz')
-        gz = Zlib::GzipReader.new(file)
-        gz.each_line do |line|
-          metadata = process_line(queue, metadata, line)
-        end
-      else
-        file.each do |line|
-          metadata = process_line(queue, metadata, line)
-        end
-      end
-    end
+    download_remote_file(object, filename)
 
-  end # def process_local_log
+    process_local_log(queue, filename)
 
-  private
-  def process_line(queue, metadata, line)
+    backup_to_bucket(object, key)
+    backup_to_dir(filename)
 
-    if /#Version: .+/.match(line)
-      junk, version = line.strip().split(/#Version: (.+)/)
-      unless version.nil?
-        metadata[:version] = version
-      end
-    elsif /#Fields: .+/.match(line)
-      junk, format = line.strip().split(/#Fields: (.+)/)
-      unless format.nil?
-        metadata[:format] = format
-      end
-    else
-      @codec.decode(line) do |event|
-        decorate(event)
-        unless metadata[:version].nil?
-          event["cloudfront_version"] = metadata[:version]
-        end
-        unless metadata[:format].nil?
-          event["cloudfront_fields"] = metadata[:format]
-        end
-        queue << event
+    delete_file_from_bucket()
+  end
+
+  def download_remote_file(remote_object, local_filename)
+    @logger.debug("S3 input: Download remove file", :remote_key => remote_object.key, :local_filename => local_filename)
+    File.open(local_filename, 'wb') do |s3file|
+      remote_object.read do |chunk|
+        s3file.write(chunk)
       end
     end
-    return metadata
+  end
 
-  end # def process_line
+  def delete_file_from_bucket
+    if @delete and @backup_to_bucket.nil?
+      object.delete()
+    end
+  end
 
-  private
-  def sincedb_read()
+  public
+  def backup_to_bucket(object, key)
+    unless @backup_to_bucket.nil?
+      backup_key = "#{@backup_add_prefix}#{key}"
+      if @delete
```

Comment:
the name of this variable might be a bit misleading, as the file gets move and not deleted.